plugins/database/postgresql/postgresql_test.go

```diff
@@ -94,6 +97,58 @@ func TestPostgreSQL_Initialize_ConnURLWithDSNFormat(t *testing.T) {
 	}
 }
 
+// Ensures we can successfully initialize and connect to a CloudSQL database
+// Requires the following:
+// - GOOGLE_APPLICATION_CREDENTIALS either JSON or path to file
+// - CONNECTION_URL to a valid Postgres instance on Google CloudSQL
+func TestPostgreSQL_Initialize_CloudGCP(t *testing.T) {
+	envConnURL := "CONNECTION_URL"
+	connURL := os.Getenv(envConnURL)
+	if connURL == "" {
+		t.Skipf("env var %s not set, skipping test", envConnURL)
+	}
+
+	credStr := dbtesting.GetGCPTestCredentials(t)
+
+	type testCase struct {
+		req dbplugin.InitializeRequest
+	}
+
+	tests := map[string]testCase{
+		"default": {
+			req: dbplugin.InitializeRequest{
+				Config: map[string]interface{}{
+					"connection_url": connURL,
+					"auth_type":      connutil.AuthTypeGCPIAM,
+				},
+				VerifyConnection: true,
+			},
+		},
+		"credentials": {
+			req: dbplugin.InitializeRequest{
+				Config: map[string]interface{}{
+					"connection_url": connURL,
+					"auth_type":      connutil.AuthTypeGCPIAM,
+					"credentials":    credStr,
+				},
+				VerifyConnection: true,
+			},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			db := new()
+			dbtesting.AssertInitializeCircleCiTest(t, db, test.req)
```

Comment:
CircleCi?

Reply:
I'm not sure - it seems like the only thing it does differently is try 5 times - we might be able to use AssertInitialize here. @vinay-gopalan?